priv/perf/apps/load_test/lib/service/faucet.ex

```diff
@@ -45,48 +69,122 @@ defmodule LoadTest.Service.Faucet do
   end
 
   def init(config) do
-    fee_wei = Keyword.fetch!(config, :fee_wei)
+    {:ok, faucet_account} = Account.new(Keyword.fetch!(config, :faucet_private_key))
+    Logger.debug("Using faucet: #{Encoding.to_hex(faucet_account.addr)}")
 
-    {faucet, eth_utxo} = get_funded_faucet_account(config)
-    Logger.debug("Using faucet: #{Encoding.to_hex(faucet.addr)}")
+    state =
+      struct!(
+        __MODULE__,
+        faucet_account: faucet_account,
+        fee: Keyword.fetch!(config, :fee_wei)
+      )
 
-    state = struct!(__MODULE__, account: faucet, fee: fee_wei, utxos: %{@eth => eth_utxo})
     {:ok, state}
   end
 
-  def handle_call(:get_faucet, _from, %__MODULE__{account: faucet} = state) do
-    {:reply, {:ok, faucet}, state}
+  def handle_call(:get_faucet, _from, %__MODULE__{faucet_account: faucet_account} = state) do
+    {:reply, {:ok, faucet_account}, state}
   end
 
-  def handle_call({:fund_child_chain, account, amount, @eth = token}, _from, %__MODULE__{account: faucet} = state) do
-    {utxo, utxo_amount} = state.utxos[token]
-    change = utxo_amount - amount - state.fee
-    if change < 0, do: raise({:error, :change_below_zero})
+  def handle_call(
+        {:fund_child_chain, receiver, amount, currency},
+        _from,
+        %__MODULE__{
+          faucet_account: faucet_account
```

Comment:
this will be much more readable, I think, if you simply ignore the pattern match - because it's the internal genserver state, you can safely assume it won't be anything else then struct.

you can then use the struct map access modifier to get the faucet account
```
def handle_call({:fund_child_chain, receiver, amount, currency}, _from, state)

`state. faucet_account `
```

Reply:
Thank you! I totally agree - I was wondering why it was done like this, but I thought I was missing some elixir-ish way of doing things 😄